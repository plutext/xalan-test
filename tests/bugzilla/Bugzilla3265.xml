<?xml version="1.0"?>

<root>
   <enumeration>
     <item value="1" />
     <item value="2" />
     <item value="3" />
   </enumeration>

   <reference ref="enumeration/item" />

</root>
