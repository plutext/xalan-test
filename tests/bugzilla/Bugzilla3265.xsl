<?xml version="1.0"?>
<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
                              xmlns:xalan="http://xml.apache.org/xalan"
                              exclude-result-prefixes="xalan">

   <!-- This stylesheet tests the evaluate extension function for the 
        case where the expression to be evaluated contains a top-level variable -->

   <xsl:variable name="this" select="root" />

   <xsl:template match="root">
      <test>
         <xsl:apply-templates/>
      </test>
   </xsl:template>

   <xsl:template match="reference">
      <xsl:variable name="var-ref" select="concat('$this/', @ref)" />
      <xsl:for-each select="xalan:evaluate($var-ref)">
         <out value="{@value}" />
      </xsl:for-each>
   </xsl:template>

</xsl:stylesheet>
