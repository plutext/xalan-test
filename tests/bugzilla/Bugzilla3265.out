<?xml version="1.0" encoding="UTF-8"?>
<test>
   
     
     
     
   

   <out value="1"/><out value="2"/><out value="3"/>

</test>